### PR TITLE
HV-1744 Use auto-expanding entries in the documentation TOC

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HV-1744

### CODE REVIEW

File: documentation/src/main/asciidoc/index.asciidoc [hunk 1]
```diff
@@ -6,10 +6,10 @@ Hardy Ferentschik; Gunnar Morling; Guillaume Smet
 :xrefstyle: full
 :anchor:
 :toc: left
-:toclevels: 3
+:toclevels: 4
 :sectnumlevels: 5
 :docinfodir: {docinfodir}
-:docinfo:
+:docinfo: shared,private
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/index.asciidoc]: This is not really necessary, but apparently it's the new syntax, so I changed this while I was at it.

**Comment by yrodiere** [comment]
I checked the rendering locally, looks good to me.

**Comment by yrodiere** [comment]
Jenkins, retest this please

**Comment by yrodiere** [comment]
@gsmet The new asciidoc "theme" is synced to Central, this PR can now be built successfully.

**Comment by yrodiere** [comment]
@gsmet Ping?

**Comment by gsmet** [comment]
Thanks for taking the time to do it, it's significantly better.

**Pull request merged**